### ISSUE TITLE
Java interoperability with Saxon breaks Ballerina build-in XML functionality

### ISSUE THREAD

**Issue by janihur** [ballerina-platform/ballerina-lang#35499]
**Description:**
I'm using Ballerina Java interoperability to work with Saxon XSLT processor (https://www.saxonica.com/). I'm getting a run-time crash when I'm using Ballerina's own XML features like:

```
var foo = xml`<a>foo</a>`;`
```

**Steps to reproduce:**

See the attached zip-file for the full source code: [p001_saxon.zip](https://github.com/ballerina-platform/ballerina-lang/files/8305370/p001_saxon.zip)

Example:
```
$ bal build && java -jar target/bin/p001_saxon.jar e032.xsl e032-input.json.xml
Resolving Maven dependencies
	Downloading dependencies into /home/jani/ballerina-ex/p001_saxon/target/platform-libs

Compiling source
	jani/p001_saxon:0.1.0

Generating executable
	warning: Detected conflicting jar files:
		'xml-apis-1.3.03.jar' dependency of 'jani/p001_saxon' conflict with 'jaxen-1.1.6.jar' dependency of 'jani/p001_saxon'
		'woodstox-core-asl-4.2.0.jar' dependency of 'jani/p001_saxon' conflict with 'ballerina-rt-2201.0.0.jar' dependency of 'jani/p001_saxon'
		'stax2-api-3.1.1.jar' dependency of 'jani/p001_saxon' conflict with 'ballerina-rt-2201.0.0.jar' dependency of 'jani/p001_saxon'
		'axiom-impl-1.2.15.jar' dependency of 'jani/p001_saxon' conflict with 'ballerina-rt-2201.0.0.jar' dependency of 'jani/p001_saxon'
		'axiom-api-1.2.15.jar' dependency of 'jani/p001_saxon' conflict with 'ballerina-rt-2201.0.0.jar' dependency of 'jani/p001_saxon'
		'commons-logging-1.2.jar' dependency of 'jani/p001_saxon' conflict with 'ballerina-rt-2201.0.0.jar' dependency of 'jani/p001_saxon'
	target/bin/p001_saxon.jar
BEGIN
{"customers":[{"id":1,"name":{"first":"Finley","last":"Goddard"}},{"id":2,"name":{"first":"Abigail","last":"Leonard"}}]}
[2022-03-18 16:34:33,787] SEVERE {b7a.log.crash} - failed to get xml as string: {ballerina/lang.xml}XMLOperationError 
io.ballerina.runtime.internal.util.exceptions.BallerinaException: failed to get xml as string: {ballerina/lang.xml}XMLOperationError
	at io.ballerina.runtime.internal.values.XmlValue.handleXmlException(XmlValue.java:153)
	at io.ballerina.runtime.internal.values.XmlNonElementItem.stringValue(XmlNonElementItem.java:195)
	at io.ballerina.runtime.internal.values.XmlNonElementItem.toString(XmlNonElementItem.java:212)
	at io.ballerina.runtime.internal.ValueUtils.createSingletonTypedesc(ValueUtils.java:180)
	at io.ballerina.runtime.internal.ValueUtils.getTypedescValue(ValueUtils.java:190)
	at io.ballerina.runtime.internal.values.XmlValue.setTypedescValue(XmlValue.java:255)
	at io.ballerina.runtime.internal.values.XmlText.<init>(XmlText.java:44)
	at io.ballerina.runtime.internal.XmlFactory.createXMLText(XmlFactory.java:383)
	at io.ballerina.runtime.internal.XmlFactory.createXMLText(XmlFactory.java:393)
	at jani.p001_saxon.0.main.main(main.bal:18)
	at jani.p001_saxon.0.$_init.$lambda$main$(p001_saxon)
	at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:611)
	at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:317)
	at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:284)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Affected Versions:**

```
$ bal --version
Ballerina 2201.0.0 (Swan Lake)
Language specification 2022R1
Update Tool 1.3.9
```

**Comment by Nadeeshan96** [comment]
This issue is solved in `Ballerina 2201.1.0 (Swan Lake)`. There it gives the below output.
```
nadeeshand@nadeeshand p001_saxon % java -jar target/bin/p001_saxon.jar e032.xsl e032-input.json.xml   
BEGIN
{"customers":[{"id":1,"name":{"first":"Finley","last":"Goddard"}},{"id":2,"name":{"first":"Abigail","last":"Leonard"}}]}
<a>foo</a>
END
```

**Comment by Nadeeshan96** [comment]
Closing because this is fixed in `Ballerina 2201.1.0 (Swan Lake)`.
